client/components/LearnerViewDirective.js

```diff
@@ -465,8 +466,20 @@ tie.directive('learnerView', [function() {
         };
 
         $scope.navigateToQuestion = function(index) {
+          // Before the questionId is changed, save it for later use.
+          var currentQuestionId = 
+            $scope.questionIds[$scope.currentQuestionIndex];
           $scope.currentQuestionIndex = index;
           var questionId = $scope.questionIds[$scope.currentQuestionIndex];
+          // Before the current code is changed to starter code for 
```

Comment:
// Whenever a user navigates to a different question,
// store their code on the scope.